src/GovUk.Education.ExploreEducationStatistics.Data.Api.Tests/Services/FrontEndServiceTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿#nullable enable
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Utils;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Data.Api.Services;
+using GovUk.Education.ExploreEducationStatistics.Data.Services.ViewModels;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace GovUk.Education.ExploreEducationStatistics.Data.Api.Tests.Services;
+
+public class FrontEndServiceTests
+{
+    private const string HttpClientName = "PublicApp";
+
+    [Fact]
+    public async Task CreateUniversalTable_ClientReturnsOk()
+    {
+        var httpMessageHandler = SetupFrontendHttpMessageHandler(HttpStatusCode.OK);
+        var httpClientFactory = HttpClientTestUtils.CreateHttpClientFactoryMock(
+            clientName: HttpClientName,
+            httpMessageHandler: httpMessageHandler.Object);
+
+        var service = BuildService(
+            httpClientFactory: httpClientFactory.Object);
+
+        var result = await service.CreateUniversalTable(
+            new TableBuilderResultViewModel(),
+            new TableBuilderConfiguration());
+
+        MockUtils.VerifyAllMocks(httpMessageHandler, httpClientFactory);
+
+        result.AssertRight();
+    }
+
+    [Fact]
+    public async Task CreateUniversalTable_ClientReturnsNotFound()
+    {
+        var httpMessageHandler = SetupFrontendHttpMessageHandler(HttpStatusCode.NotFound);
+        var httpClientFactory = HttpClientTestUtils.CreateHttpClientFactoryMock(
+            clientName: HttpClientName,
+            httpMessageHandler: httpMessageHandler.Object);
+
+        var service = BuildService(
+            httpClientFactory: httpClientFactory.Object);
+
+        var result = await service.CreateUniversalTable(
+            new TableBuilderResultViewModel(),
+            new TableBuilderConfiguration());
+
+        MockUtils.VerifyAllMocks(httpMessageHandler, httpClientFactory);
+
+        result.AssertNotFound();
+    }
+
+    [Fact]
+    public async Task CreateUniversalTable_ClientReturnsError()
+    {
+        var httpMessageHandler = SetupFrontendHttpMessageHandler(HttpStatusCode.BadRequest);
+        var httpClientFactory = HttpClientTestUtils.CreateHttpClientFactoryMock(
+            clientName: HttpClientName,
+            httpMessageHandler: httpMessageHandler.Object);
+
+        var service = BuildService(
+            httpClientFactory: httpClientFactory.Object);
+
+        var result = await service.CreateUniversalTable(
+            new TableBuilderResultViewModel(),
+            new TableBuilderConfiguration());
+
+        MockUtils.VerifyAllMocks(httpMessageHandler, httpClientFactory);
+
+        result.AssertStatusCodeResult(HttpStatusCode.InternalServerError);
```

Comment:
I'm not sure that `HttpResponseMessageTestExtensions` makes sense here because `result` isn't a `HttpResponseMessage` (this isn't an integration test).

I've made this a bit nicer now though with an extension, so it's `result.AssertInternalServerError();`